### PR TITLE
 Refactor eglot--maybe-activate-managed-mode 

### PR DESCRIPTION
**Pull request by rbrtb**
Fixes https://github.com/joaotavora/eglot/issues/824

### CODE REVIEW

File: eglot.el [hunk 1]
```diff
@@ -1723,9 +1724,6 @@ If it is activated, also signal textDocument/didOpen."
       (eglot--managed-mode)
       (eglot--signal-textDocument/didOpen))))
 
-(add-hook 'find-file-hook 'eglot--maybe-activate-editing-mode)
```
Comment (joaotavora) [inline eglot.el:1726]: What makes you think these two lines are unneeded?  As far as i remember they are the basis for the most basic of Eglot underpinnings: the capability to automatically enable eglot when visiting files for a given project.  

Not everyone uses `eglot-ensure` (I don't, for one).

Comment (rbrtb) [reply eglot.el:1726]: How about an option to toggle it?

**Comment by joaotavora** [comment]
OK.  But instead of adding structural changes to `eglot.el`, how about describing exactly the use case that you're thinking of solving?

Earlier, I had the impression that your use case concerns usage of `eglot-ensure` in a specific scenario, which isn't perfectly described (as in a formal bug report that I myself can witness).  So why not do one of these things, or both:

* describe the minimal reproducible example that shows evidence of the bug
* think about touching `eglot-ensure`?

**Comment by rbrtb** [comment]
When magit created a non-file buffer(such as `magit-blob-previous` on a git tracked file), 
it will temporarily set `buffer-file-name' for the usage of `normal-mode`:
```
(let ((buffer-file-name magit-buffer-file-name)
          (after-change-major-mode-hook
           (remq 'global-diff-hl-mode-enable-in-buffers
                 after-change-major-mode-hook)))
      (normal-mode t))
```
Thereby `eglot--managed-mode` is enabled on that buffer. 
Then when you move around in that buffer, eldoc is triggered, and there's error: `(wrong-type-argument arrayp nil)`.

**Comment by rbrtb** [comment]
And if you think current changes are too structural, perhaps consider adding an option to toggle automanage? https://github.com/joaotavora/eglot/pull/942/commits/6900927f7433adede372eeee7631dc1132fa8711

**Comment by joaotavora** [comment]
> it will temporarily set buffer-file-name for the usage of normal-mode:

Magic lying then, right? It says there is a supporting file, but there is none. I can't tell if this is the problem from your short description. I also have no idea what normal mode is.

Also your current implementation breaks all Eglot tests.

Adding a toggle to something as fundamental as Eglot's most basic working principle sounds more complicated than perhaps asking Magic matters why they do those things.

**Comment by rbrtb** [comment]
Well I agree that it's more of a magit problem, so I'm closing this.

**Pull request closed without merging**